File: garden_advice.js
// TODO: Examples of possible features to add:
// - Add detailed comments explaining each block of code.
// - Refactor the code into functions for better readability and modularity.
// - Store advice in an object for multiple plants and seasons.
// - Suggest plants that thrive in the given season.

// Garden Advice App
// This program gives gardening tips based on the selected season and plant type.
// It uses functions for modularity, an object for advice storage,
// and provides plant recommendations for each season.

// Object containing advice for each season and plant type
const adviceData = {
  spring: {
    flower: "Prepare soil and plant new bulbs for colorful blooms.",
    vegetable: "Start planting early crops like lettuce and peas.",
    herb: "Plant herbs in pots for easy indoor care."
  },
  summer: {
    flower: "Water regularly and provide shade during hot days.",
    vegetable: "Harvest ripe crops and check for pests.",
    herb: "Trim herbs frequently to encourage new growth."
  },
  autumn: {
    flower: "Prune dead flowers and collect seeds for next season.",
    vegetable: "Clear out harvested plants and enrich the soil.",
    herb: "Dry herbs for storage and prepare them for cooler weather."
  },
  winter: {
    flower: "Protect flowers from frost with garden covers.",
    vegetable: "Focus on soil care and plan next season’s crops.",
    herb: "Move potted herbs indoors to keep them warm."
  }
};

// Object recommending plants for each season
const plantRecommendations = {
  spring: ["tulips", "peas", "mint"],
  summer: ["roses", "tomatoes", "basil"],
  autumn: ["marigolds", "carrots", "thyme"],
  winter: ["pansies", "cabbage", "rosemary"]
};

// Function to get user input for season
function getSeason() {
  const season = prompt("Enter a season (spring, summer, autumn, winter):").toLowerCase();
  return ["spring", "summer", "autumn", "winter"].includes(season) ? season : "summer";
}

// Function to get user input for plant type
function getPlantType() {
  const type = prompt("Enter a plant type (flower, vegetable, herb):").toLowerCase();
  return ["flower", "vegetable", "herb"].includes(type) ? type : "flower";
}

// Function to get gardening advice based on season and plant type
function getAdvice(season, plantType) {
  if (adviceData[season] && adviceData[season][plantType]) {
    return adviceData[season][plantType];
  } else {
    return "No advice available for this selection.";
  }
}

// Function to recommend plants based on the selected season
function recommendPlants(season) {
  const plants = plantRecommendations[season];
  if (plants) {
    console.log(`Recommended plants for ${season}:`);
    plants.forEach(plant => console.log("- " + plant));
  } else {
    console.log("No plant recommendations available for this season.");
  }
}

// Main function to control the menu flow
function main() {
  while (true) {
    const choice = prompt(
      "Garden Advice Menu:\n1. Get gardening advice\n2. Recommend plants\n3. Exit\nEnter your choice:"
    );

    if (choice === "1") {
      const season = getSeason();
      const plantType = getPlantType();
      console.log("\nGardening Advice:");
      console.log(getAdvice(season, plantType));
    } else if (choice === "2") {
      const season = getSeason();
      recommendPlants(season);
    } else if (choice === "3") {
      console.log("Goodbye.");
      break;
    } else {
      console.log("Invalid choice. Please try again.");
    }
  }
}

// Run the program
main();

File: garden_advice.py

# # TODO: Examples of possible features to add:
# # - Add detailed comments explaining each block of code.
# # - Refactor the code into functions for better readability and modularity.
# # - Store advice in a dictionary for multiple plants and seasons.
# # - Recommend plants based on the entered season.


# Garden Advice App
# This program gives gardening tips
#  based on the selected season and plant type.
# It uses functions for modularity, a dictionary for advice storage,
# and provides plant recommendations for each season.

# Dictionary containing advice for each season and plant type
advice_data = {
    "spring": {
        "flower": "Prepare soil and plant new bulbs for colorful blooms.",
        "vegetable": "Start planting early crops like lettuce and peas.",
        "herb": "Plant herbs in pots for easy indoor care."
    },
    "summer": {
        "flower": "Water regularly and provide shade during hot days.",
        "vegetable": "Harvest ripe crops and check for pests.",
        "herb": "Trim herbs frequently to encourage new growth."
    },
    "autumn": {
        "flower": "Prune dead flowers and collect seeds for next season.",
        "vegetable": "Clear out harvested plants and enrich the soil.",
        "herb": "Dry herbs for storage and prepare them for cooler weather."
    },
    "winter": {
        "flower": "Protect flowers from frost with garden covers.",
        "vegetable": "Focus on soil care and plan next season’s crops.",
        "herb": "Move potted herbs indoors to keep them warm."
    }
}

# Dictionary recommending which plants are best for each season
plant_recommendations = {
    "spring": ["tulips", "peas", "mint"],
    "summer": ["roses", "tomatoes", "basil"],
    "autumn": ["marigolds", "carrots", "thyme"],
    "winter": ["pansies", "cabbage", "rosemary"]
}


def get_season():
    """Display a menu and return the chosen season."""
    print("\nSelect a season:")
    print("1. Spring")
    print("2. Summer")
    print("3. Autumn")
    print("4. Winter")
    choice = input("Enter the number of your choice: ")

    if choice == "1":
        return "spring"
    elif choice == "2":
        return "summer"
    elif choice == "3":
        return "autumn"
    elif choice == "4":
        return "winter"
    else:
        print("Invalid choice. Defaulting to 'summer'.")
        return "summer"


def get_plant_type():
    """Display a menu and return the chosen plant type."""
    print("\nSelect a plant type:")
    print("1. Flower")
    print("2. Vegetable")
    print("3. Herb")
    choice = input("Enter the number of your choice: ")

    if choice == "1":
        return "flower"
    elif choice == "2":
        return "vegetable"
    elif choice == "3":
        return "herb"
    else:
        print("Invalid choice. Defaulting to 'flower'.")
        return "flower"


def get_advice(season, plant_type):
    """Retrieve gardening advice from
      the dictionary based on the season and plant type."""
    # Get advice from dictionary, or a default message if not found
    season_data = advice_data.get(season, {})
    return season_data.get(
        plant_type, "No advice available for this selection.")


def recommend_plants(season):
    """Suggest suitable plants to grow based on the selected season."""
    plants = plant_recommendations.get(season, [])
    if plants:
        print("\nRecommended plants for", season.capitalize() + ":")
        for plant in plants:
            print("-", plant.capitalize())
    else:
        print("No plant recommendations available for this season.")


def main():
    """Main menu loop for the garden advice program."""
    print("Welcome to the Garden Advice App")

    while True:
        # Display main menu
        print("\nMenu:")
        print("1. Get gardening advice")
        print("2. Recommend plants for the season")
        print("3. Exit")

        choice = input("Enter your choice: ")

        if choice == "1":
            season = get_season()
            plant_type = get_plant_type()
            print("\nGardening Advice:")
            print(get_advice(season, plant_type))
        elif choice == "2":
            season = get_season()
            recommend_plants(season)
        elif choice == "3":
            print("Goodbye.")
            break
        else:
            print("Invalid choice. Please try again.")


if __name__ == "__main__":
    main()
